### DISCUSSION TITLE
How to assign block id for embedded line in GMSH

### DISCUSSION THREAD

**Discussion by rksin8** [idaholab/moose#17958] [Q&A Meshing] [answered]
Dear moose users,

I have a mesh that contains an embedded line. I want to assign properties based on block ID but when I open this mesh in ParaView, I see only one block ID for the surface. How can I create a block ID for the line as well? 

Input for geo file:
```
Mesh.Algorithm = 1;
Mesh.Optimize = 1;
lc=50;
lc2=10;
Point(0)={0,0,0,lc};
Point(1)={500,0,0,lc};
Point(2)={500,500,0,lc};
Point(3)={0,500,0,lc};
Point(4)={100,250,0,lc2};
Point(5)={400,250,0,lc2};
Point(6)={175,250,0,lc2};
Point(7)={325,250,0,lc2};
Line (1) ={0,1};
Line (2) ={1,2};
Line (3) ={2,3};
Line (4) ={3,0};
Line (5) ={4,6};
Line (6) ={6,7};
Line (7) ={7,5};
Line Loop(1) ={1,2,3,4};
Ruled Surface(1) = {1};
Line {5} In Surface {1};
Line {6} In Surface {1};
Line {7} In Surface {1};
Point {6} In Surface {1};
Point {7} In Surface {1};
Physical Surface(1) = {1};
Physical Line("fracture") = {6, 5, 7};
Physical Line("left") = {4};
Physical Line("right") = {2};
Physical Line("top") = {3};
Physical Line("bottom") = {1};
Physical Point("injection_well") = {6};
Physical Point("production_well") = {7};
```
Regards,
Ranjeet

**Comment by GiudGiud** [comment]
@lindsayad if you know.
Otherwise, Ranjeet, if you get the answer from a gmsh forum, please post the answer here to help others. 

Thanks

**Reply by rksin8** [reply]
Thank you for your reply. now we have a solution.

**Comment by lindsayad** [comment, marked as answer]
@rksin8 If you want to create a lower dimensional block, you need to make sure that the Physical name contains the string `lower_dimensional_block`. So if you want "fracture" to be a lower dimensional block, then you would do this:
```
Physical Line("fracture_lower_dimensional_block") = {6, 5, 7};
```

**Reply by rksin8** [reply]
It worked! Thank you so much.

**Reply by hugary1995** [reply]
Awesome! I was just about to ask the same question.
